Add Liquity protocol

edmulraney opened the pull request:
Protocol: https://www.liquity.org/
Docs: https://docs.liquity.org/ (user focused) and https://github.com/liquity/dev (developer focused)

This connector adds support for interacting with Liquity protocol. Every function from Liquity has been proxied in the connector.

Users can:

### Perform Trove operations:
- Create, adjust and close a Trove

### Perform Stability Pool operations
- Deposit and withdraw LUSD from the Stability Pool
- Claim LQTY gains

### Perform LQTY staking operations:
- Stake and unstake LQTY
- Claim ETH and LUSD gains from Staking

edmulraney committed:
add liquity connector

edmulraney committed:
add recommended solidity version

edmulraney committed:
add test for instadapp event return values. address PR comments:
    - add setUints to all applicable functions
    - use uint256 rather than uint for event data

edmulraney committed:
include all gains from staking/stability-pool when performing operations on them

edmulraney committed:
add tests to ensure getId/setIds all work

edmulraney committed:
log the eth and lqty gains from stability pool operations. general tidy up

edmulraney committed:
Merge pull request #1 from edmulraney/liquity-connector

Instadapp integration: add Liquity connector

edmulraney committed:
Merge branch 'main' of github.com:Instadapp/dsa-connectors

KaymasJain committed:
restructuring & updated function till repay

KaymasJain committed:
updated all the functions other adjust

KaymasJain committed:
liquity updates done

KaymasJain committed:
added struct on adjust setIds

KaymasJain committed:
updated the event with data from struct

edmulraney committed:
fix broken tests from instadapp PR

edmulraney committed:
Merge pull request #3 from liquity/liquity-updates

Fix tests after Instadapp team changes

edmulraney committed:
Merge branch 'main' of github.com:liquity/dsa-connectors

edmulraney committed:
Merge remote-tracking branch 'upstream/main'

KaymasJain commented on contracts/mainnet/connectors/liquity/main.sol:
```diff
@@ -0,0 +1,458 @@
+        uint setWithdrawId,
+        uint setEthGainId,
+        uint setLqtyGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

KaymasJain commented on contracts/mainnet/connectors/liquity/main.sol:
```diff
@@ -0,0 +1,458 @@
+    function stabilityMoveEthGainToTrove(
+        address upperHint,
+        address lowerHint
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

KaymasJain commented on contracts/mainnet/connectors/liquity/main.sol:
```diff
@@ -0,0 +1,458 @@
+        uint setStakeId,
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

KaymasJain commented on contracts/mainnet/connectors/liquity/main.sol:
```diff
@@ -0,0 +1,458 @@
+        uint setUnstakeId,
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

KaymasJain commented on contracts/mainnet/connectors/liquity/main.sol:
```diff
@@ -0,0 +1,458 @@
+    function claimStakingGains(
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

edmulraney committed:
Allow non-payable functions to be composable with payable functions

Make all functions payable so they can be composed in a larger spell which might use msg.value (normally non-payable is rejected)

Co-authored-by: Samyak Jain <34437877+KaymasJain@users.noreply.github.com>

edmulraney committed:
Allow non payable functions to be composable with payable functions

Make all functions payable so they can be composed in a larger spell which might use msg.value (normally non-payable is rejected)

Co-authored-by: Samyak Jain <34437877+KaymasJain@users.noreply.github.com>

edmulraney committed:
Allow non-payable functions to be composable with payable functions

Make all functions payable so they can be composed in a larger spell which might use msg.value (normally non-payable is rejected)

Co-authored-by: Samyak Jain <34437877+KaymasJain@users.noreply.github.com>

edmulraney committed:
Allow non-payable functions to be composable with payable functions

Make all functions payable so they can be composed in a larger spell which might use msg.value (normally non-payable is rejected)

Co-authored-by: Samyak Jain <34437877+KaymasJain@users.noreply.github.com>

edmulraney committed:
make all functions payable so they can be composed with payable functions

edmulraney committed:
Merge remote-tracking branch 'upstream/main'
